src/debugAdapter/goDebug.ts

```diff
@@ -751,6 +752,23 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	protected setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments): void {
+		log('SetBreakPointsRequest');
+		if (!this.continueRequestRunning) {
+			this.setBreakPoints(response, args);
+		} else {
+			this.skipStopEventOnce = true;
+			this.delve.callPromise('Command', [{ name: 'halt' }]).then(() => {
+				return this.setBreakPoints(response, args);
+			}, err => {
+				this.sendErrorResponse(response, 2002, 'Failed to set breakpoint: "{e}"', { e: err.toString() });
+				logError(err);
+			}).then(() => {
+				return this.continue();
```

Comment:
I believe this will get called even when the Halt request fails.
Is that what we want?
I assumed that we want to `continue` only if halting was successful

Reply:
Fixed